sqlalchemy-stubs/sql/type_api.pyi

```diff
@@ -109,8 +110,9 @@ class Variant(TypeDecorator[_T]):
     def coerce_compared_value(self: _VT, operator: Any, value: Any) -> Union[_VT, TypeEngine[_T]]: ...
     def load_dialect_impl(self, dialect: Dialect) -> TypeEngine[Any]: ...
     def with_variant(self, type_: TypeEngine[_U], dialect_name: str) -> Variant[_U]: ...
+    # Signature of comparator factory incompatible with supertype TypeEngine
     @property
-    def comparator_factory(self) -> Type[Any]: ...
+    def comparator_factory(self) -> Type[Any]: ...  # type: ignore
```

Comment:
There is some inconsistency in how `comparator_factory` is defined in various classes, sometimes it is a read-only property, but sometimes it is a writeable attribute. Maybe these properties are actually settable so we can replace them with just normal attributes?

Reply:
For `Variant` and `TypeDecorator`, `comparator_factory` is calculated from the implementation, so it has to be a read-only property. We could type it as read/write, but I feel wrong doing that.